lib/diagnostic.js

```diff
@@ -8,40 +8,129 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-const min = function min() {
+
+let arr = [1,2,3];
+
+
+let callbackMin = function (element) {
+
+  let firstElement = arr[0];
+
+  for (let i = 0; i < array.length; i++) {
+    let result =   array[i];
+
+if (firstElement < result) {
+  return firstElement;
+}
+
+  }
+
+};
+
+const min = function min(array, callback) {
+
 
```

Comment:
You're on the right track with this. Remember callbacks with `reduce` are dealing with previous and current values. Here is one solution:

```
const min = function min(previousValue, currentValue) {
  return previousValue < currentValue ? previousValue : currentValue;
};
```